core/metacling/src/TClingMethodArgInfo.cxx

```diff
@@ -40,12 +40,18 @@ the Clang C++ compiler, not CINT.
 
 #include <string>
 
+TClingMethodArgInfo::TClingMethodArgInfo(cling::Interpreter *interp, const TClingMethodInfo* mi) : TClingGenericInfo(mi->GetMethodDecl()), fInterp(interp), fIdx(-1) {}
+
 bool TClingMethodArgInfo::IsValid() const
 {
-   if (!fMethodInfo || !fMethodInfo->IsValid()) {
+   // Calling the base class implementation is unsafe because we override the
+   // GetDecl which it forwads to. That implementation depends on fIdx which is
+   // used to determine validity.
+   if (!fDecl)
```

Comment:
Are the fDecl and fMethodInfo really interchangeable?

Reply:
In this particular use I believe so. I have traced the use pattern and it seems a decl is sufficient. Let's try to appease the bots first ;)

Reply:
Fair enough.  This difference was a plausible cause of the problems.